packages/gatsby-image/src/index.js

```diff
@@ -216,7 +217,14 @@ class Image extends React.Component {
           this.props.onStartLoad({ wasCached: imageInCache })
         }
 
-        this.setState({ isVisible: true, imgLoaded: imageInCache })
+        // imgCached and imgLoaded must update after isVisible,
+        // more details in this comments related commit
```

Comment:
it's probably a good way to put the comment here as well just so we don't forget it in the long run.

Reply:
Well, that was the point of the comment there, so it could be looked up via git blame, no one should have a reason to remove the comment knowing it's giving a reason for why the related code is how it is.

I'll update it to not rely on that, it's just longer, which I thought would be preferred to avoid.